### PR TITLE
Initial bot settings

### PR DESCRIPTION
**Pull request by mdrgoncalves**
Changes throughout the repository.

- Set initial bot settings for its full operation.
- Created commands: Hello.

### CODE REVIEW

File: src/commands/FxTweet.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      name: 'tweet',
+      description: 'Twitter URL',
+      type: ApplicationCommandOptionType.String,
+      required: true,
+    },
+  ],
+  run: (
+    client: Client,
+    interaction: CommandInteraction,
+  ): Promise<InteractionResponse<boolean>> => {
```
Comment (lcsmuller) [inline src/commands/FxTweet.ts]: Nit: `Command` already infers callback's types!
```suggestion
import { ApplicationCommandType, ApplicationCommandOptionType } from 'discord.js';

import { Command } from '../interfaces/Command';

export const FxTweet: Command = {
  name: 'fxtweet',
  description: 'Fix a tweet video embed',
  type: ApplicationCommandType.ChatInput,
  options: [
    {
      name: 'tweet',
      description: 'Twitter URL',
      type: ApplicationCommandOptionType.String,
      required: true,
    },
  ],
  run: (_, interaction) => {
```

File: src/interfaces/Command.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { CommandInteraction, ChatInputApplicationCommandData, Client } from "discord.js";
+
+export interface Command extends ChatInputApplicationCommandData {
+    run: (client: Client, interaction: CommandInteraction) => void;
```
Comment (lcsmuller) [inline src/interfaces/Command.ts]: Should it return a `Promise<InteractionResponse<boolean>>` instead?

File: src/commands/Hello.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  ): Promise<InteractionResponse<boolean>> => {
+    const username = interaction.user?.username;
+    const content = `Ara ara... Hi, ${username}-kun!`;
+
+    return interaction.reply({
+      ephemeral: true,
+      content,
+    });
+  },
+};
```
Comment (lcsmuller) [inline src/commands/Hello.ts]: ditto, also we could turn this into `as const satisfies` and achieve better typing..
```suggestion
import { ApplicationCommandType } from 'discord.js';

import { Command } from '../interfaces/Command';

export const Hello = {
  name: 'hello',
  description: 'Returns a greeting',
  type: ApplicationCommandType.ChatInput,
  run: (_, interaction) => {
    const username = interaction.user?.username;
    const content = `Ara ara... Hi, ${username}-kun!`;

    return interaction.reply({
      ephemeral: true,
      content,
    });
  },
} as const satisfies Command;
```

File: src/commands/Hello.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+export const Hello: Command = {
+  name: 'hello',
+  description: 'Returns a greeting',
+  type: ApplicationCommandType.ChatInput,
+  run: (
+    client: Client,
+    interaction: CommandInteraction,
+  ): Promise<InteractionResponse<boolean>> => {
+    const username = interaction.user?.username;
```
Comment (lcsmuller) [inline src/commands/Hello.ts]: can user ever be undefined here? we may want to throw an error in that case

File: src/listeners/interactionCreate.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { Client, CommandInteraction, Interaction } from 'discord.js';
+import { Commands } from '../Commands';
+
+export default (client: Client): void => {
+  client.on('interactionCreate', async (interaction: Interaction) => {
+    if (interaction.isCommand()) {
+      await handleSlashCommand(client, interaction);
+    }
+  });
+};
```
Comment (lcsmuller) [inline src/listeners/interactionCreate.ts]: Nit: type is inferred automatically, `async` may be omitted
```suggestion
export default (client: Client): void => {
  client.on('interactionCreate', interaction => {
    if (interaction.isCommand()) {
      return handleSlashCommand(client, interaction);
    }
  });
};
```

File: src/listeners/interactionCreate.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  const slashCommand = Commands.find(
+    command => command.name === interaction.commandName,
+  );
+  if (!slashCommand) {
+    interaction.followUp({ content: 'An error has occurred' });
+    return;
+  }
+
+  slashCommand.run(client, interaction);
+};
```
Comment (lcsmuller) [inline src/listeners/interactionCreate.ts]: Unused `async`, but `interaction.followUp` seems to return a Promise:
```suggestion
const handleSlashCommand = (
  client: Client,
  interaction: CommandInteraction,
): Promise<void> => {
  const slashCommand = Commands.find(
    command => command.name === interaction.commandName,
  );
  if (!slashCommand) {
    return interaction.followUp({ content: 'An error has occurred' });
  }

  slashCommand.run(client, interaction);
};
```

File: src/commands/FxTweet.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  CommandInteraction,
+  Client,
+  ApplicationCommandType,
+  ApplicationCommandOptionType,
+  InteractionResponse,
+} from 'discord.js';
+
+import { Command } from '../interfaces/Command';
+
+export const FxTweet: Command = {
```
Comment (lcsmuller) [inline src/commands/FxTweet.ts]: Up to you, but using `satisfies as Command` should provide more accurate typing when hovering the mouse over it:
```ts
const FxTweet: Command
// vs
const FxTweet: {
    name: string;
    description: string;
    type: ApplicationCommandType.ChatInput;
    options: {
        name: string;
        description: string;
        type: ApplicationCommandOptionType.String;
        required: true;
    }[];
    run: (_: Client<...>, interaction: CommandInteraction<...>) => Promise<...>;
}
```

File: src/commands/FxTweet.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import {
```
Comment (lcsmuller) [inline src/commands/FxTweet.ts]: Nit: These are all types, so may be imported as such:
```suggestion
import type {
```

File: src/commands/Hello.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import {
```
Comment (lcsmuller) [inline src/commands/Hello.ts]: ditto

File: src/commands/Hello.ts [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  CommandInteraction,
+  Client,
+  ApplicationCommandType,
+  InteractionResponse,
+} from 'discord.js';
+
+import { Command } from '../interfaces/Command';
```
Comment (lcsmuller) [inline src/commands/Hello.ts]: ditto

File: src/index.ts [hunk 11]
```diff
@@ -1,20 +1,17 @@
-import { Client } from 'discord.js';
+import { Client } from "discord.js";
```
Comment (lcsmuller) [inline src/index.ts:1]: ditto

File: src/listeners/interactionCreate.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import { Client, CommandInteraction, Interaction } from 'discord.js';
+import { Commands } from '../Commands';
```
Comment (lcsmuller) [inline src/listeners/interactionCreate.ts]: ditto

File: src/listeners/ready.ts [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import { Client, ActivityType } from 'discord.js';
+import { Commands } from '../Commands';
```
Comment (lcsmuller) [inline src/listeners/ready.ts]: ditto

**Review by Guilherme-SG — APPROVED** [review]

**Pull request merged**